dlhub_sdk/client.py

```diff
@@ -276,21 +276,32 @@ def run(self, name: str, inputs: Any, parameters: Optional[Dict[str, Any]] = Non
                 - error_message: Exception traceback
             If neither, the output of the function
         """
-
-        if name not in self.fx_cache:
+        # if identifier is a DOI
+        if identifier.startswith("10.") or identifier.startswith("https://doi.org/"):
```

Comment:
what if it's not a properly formatted DOI? Please throw an error to let the user know that their input was incorrect. Otherwise, this will probably hang or yield an uninformative error

Reply:
an error will be thrown if the DOI is not found (correct or not), which will at least tell them where to look. would you prefer a specific check for valid formatting?